libr/util/str.c

```diff
@@ -2928,3 +2929,21 @@ R_API char* r_str_highlight(char *str, const char *word, const char *color) {
 	}
 	return strdup (o);
 }
+
+R_API int r_str_scpy (char *dst, const char *src, int count) {
+  if (count <= 0) {
+    return -E2BIG;
+  }
+  int ret = 0;
+  while (count) {
+    if (!(dst[ret] = src[ret])) {
+      return ret;
+    }
+    ret++;
+    count--;
+  }
+  if (ret) {
+    dst[ret-1] = '\0';
+  }
+  return -E2BIG;
```

Comment:
The intention of `E2BIG` return value is to make it easy to check if overflow occurs. Sometimes `count` is calculated e.g. `r_str_scpy (dst + i, src, sizeof dst - i);`. In this way, you do not need `r_str_scpy (...) >= sizeof dst -i)` but `r_str_scpy (...) < 0` to check if overflow occurs.